source/standard_retry_strategy.c

```diff
@@ -350,26 +365,26 @@ static int s_standard_retry_strategy_schedule_retry(
 static int s_standard_retry_strategy_record_success(struct aws_retry_token *token) {
     struct retry_bucket_token *impl = token->impl;
 
-    AWS_FATAL_ASSERT(!aws_mutex_lock(&impl->strategy_bucket->partition_lock) && "mutex lock failed");
+    AWS_FATAL_ASSERT(!aws_mutex_lock(&impl->strategy_bucket->synced_data.partition_lock) && "mutex lock failed");
     AWS_LOGF_DEBUG(
         AWS_LS_IO_STANDARD_RETRY_STRATEGY,
         "token_id=%p: partition=" PRInSTR
         ": recording successful operation and adding %zu units of capacity back to the bucket.",
         (void *)token,
         AWS_BYTE_CURSOR_PRI(impl->strategy_bucket->partition_id_cur),
         impl->last_retry_cost);
-    size_t capacity_payback = impl->strategy_bucket->current_capacity + impl->last_retry_cost;
+    size_t capacity_payback = impl->strategy_bucket->synced_data.current_capacity + impl->last_retry_cost;
```

Comment:
Question that I didn't think of during the first review: if an operation fails multiple times, and we re-use the same token for that operation multiple times, is it okay that the capacity only gets paid back for the last retry cost of the token, and not all of the failures?  (Seems intentional, just wanted to check.)

Reply:
it only ever gets paid back for success. You should reuse the token for multiple retries, that's what they're for. The accounting is done when attempting to retry or to record success. This is fine.